From 6da263921b9b9cd093706575dc901ac4511901c4 Mon Sep 17 00:00:00 2001
From: Ivan Iziumov <iiziumov@anaconda.com>
Date: Mon, 2 Mar 2026 10:16:30 +0100
Subject: [PATCH] remove problematic win test

Rebased onto 1.30.0: disable only onnxruntime_shared_lib_dlopen_test (and its MSVC debugger property); the cpuinfo dlopen test added in 1.30 stays enabled.
---
 cmake/onnxruntime_unittests.cmake | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/cmake/onnxruntime_unittests.cmake b/cmake/onnxruntime_unittests.cmake
index 1974b5d..365ae15 100644
--- a/cmake/onnxruntime_unittests.cmake
+++ b/cmake/onnxruntime_unittests.cmake
@@ -2043,10 +2043,10 @@ endif()
   # shared lib
   if (onnxruntime_BUILD_SHARED_LIB)
     if(WIN32)
-      onnxruntime_add_executable(onnxruntime_shared_lib_dlopen_test ${ONNXRUNTIME_SHARED_LIB_TEST_SRC_DIR}/dlopen_main.cc)
-      add_dependencies(onnxruntime_shared_lib_dlopen_test ${all_dependencies} onnxruntime)
-      add_test(NAME onnxruntime_shared_lib_dlopen_test COMMAND onnxruntime_shared_lib_dlopen_test WORKING_DIRECTORY $<TARGET_FILE_DIR:onnxruntime_shared_lib_dlopen_test>)
-      set_target_properties(onnxruntime_shared_lib_dlopen_test PROPERTIES FOLDER "ONNXRuntimeTest")
+      # onnxruntime_add_executable(onnxruntime_shared_lib_dlopen_test ${ONNXRUNTIME_SHARED_LIB_TEST_SRC_DIR}/dlopen_main.cc)
+      # add_dependencies(onnxruntime_shared_lib_dlopen_test ${all_dependencies} onnxruntime)
+      # add_test(NAME onnxruntime_shared_lib_dlopen_test COMMAND onnxruntime_shared_lib_dlopen_test WORKING_DIRECTORY $<TARGET_FILE_DIR:onnxruntime_shared_lib_dlopen_test>)
+      # set_target_properties(onnxruntime_shared_lib_dlopen_test PROPERTIES FOLDER "ONNXRuntimeTest")
 
       if(onnxruntime_cpuinfo_library_type STREQUAL "STATIC_LIBRARY")
         onnxruntime_add_shared_library(
@@ -2080,10 +2080,10 @@ endif()
           PROPERTIES FOLDER "ONNXRuntimeTest")
       endif()
 
-      if (MSVC)
-        # set VS debugger working directory to the test program's directory
-        set_target_properties(onnxruntime_shared_lib_dlopen_test PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY $<TARGET_FILE_DIR:onnxruntime_shared_lib_dlopen_test>)
-      endif()
+      # if (MSVC)
+      #   # set VS debugger working directory to the test program's directory
+      #   set_target_properties(onnxruntime_shared_lib_dlopen_test PROPERTIES VS_DEBUGGER_WORKING_DIRECTORY $<TARGET_FILE_DIR:onnxruntime_shared_lib_dlopen_test>)
+      # endif()
     endif()
     onnxruntime_add_static_library(onnxruntime_mocked_allocator ${TEST_SRC_DIR}/util/test_allocator.cc)
     target_include_directories(onnxruntime_mocked_allocator PUBLIC ${TEST_SRC_DIR}/util/include)
